main.py

```diff
@@ -670,7 +671,13 @@ def run(
                     f"Adding {len(mgr.pools_to_add_from_contracts)} new pools from contracts, "
                     f"{len(mgr.pool_data)} total pools currently exist. Current block: {current_block}."
                 )
-                async_update_pools_from_contracts(mgr, current_block, logging_path)
+                try:
+                    async_update_pools_from_contracts(mgr, current_block, logging_path)
+                    failed_async_calls = 0
+                except Exception:
+                    failed_async_calls += 1
+                    if failed_async_calls >= 3:
+                        raise Exception(f"[main run.py] async_update_pools_from_contracts failed. List of failed pools: {mgr.pools_to_add_from_contracts}")
```

Comment:
How are all of these changes related to the issue that this PR is attempting to address?

Reply:
This fixes a different issue I noticed that caused trades including Solidly pools to fail, specifically when using the startup flag run_data_validator.